### PR TITLE
feat(l10n): check if translated urls are valid

### LINKED ISSUE #2763: Find if URLs were changed during translation in l10n
Following up from #2742  and comments from @vladikoff :
Not related to this PR, but we need a way to find that HREF was changed:
i.e 
```
账号已存在。<a href="/signin">登录</a>
became
账号已存在。<a href="http://bing.com">登录</a>
```
We need to figure out a way to find if URLs were changed during translation in l10n.

### PR DESCRIPTION
**Pull request by TDA**
Fixes #2763
Generate the set of valid urls from .pot files
Check if the translated files actually contain only these urls
@vladikoff 
P.S: Vlad, Find out the easter egg :P

Depends on #2742

### CODE REVIEW

File: grunttasks/l10n-translate-verify.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            grunt.log.error('Found an url which should not be there', url, src);
+            hasErrors = hasErrors + 1;
+          }
+        });
+      }
+    });
+    if (hasErrors !== 0) {
+      grunt.fail.warn('Found ' + hasErrors + ' ' + grunt.util.pluralize(hasErrors, 'error/errors') + ' in ' + fileArray.length + ' files');
+    } else {
+      grunt.log.writeln('Checked ' + fileArray.length + ' files for valid hrefs and found nothbing wrong :D');
```
Comment (pdehaan) [inline grunttasks/l10n-translate-verify.js]: typo: "nothing"

Comment (TDA) [inline grunttasks/l10n-translate-verify.js]: its nothbing, not-having-bing, not-h-bing. y u do this to me, Peter? :japanese_goblin:

File: grunttasks/l10n-translate-verify.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    // iterate through eact .pot file and extract the urls
+    this.files[0].src.forEach(function (src) {
+      var contents = grunt.file.read(src);
+      if (urlRegex.test(contents)) {
+        Array.prototype.push.apply(urlArray, contents.match(urlRegex));
+      }
+    });
+    for (var i = 0; i < urlArray.length; i++) {
+      urlArray[i] = stripSpecialChars(urlArray[i]);
+    }
```
Comment (pdehaan) [inline grunttasks/l10n-translate-verify.js]: You may be able to optimize this down to something like this pseudo-mess:

```js
urlArray = urlArray.map(stripSpecialChars);
```

File: grunttasks/l10n-translate-verify.js (hunk 1 above)
Comment (pdehaan) [inline grunttasks/l10n-translate-verify.js]: @TDA: Deep down, maybe I'm just a bad person.

File: grunttasks/l10n-translate-verify.js (hunk 2 above)
Comment (TDA) [inline grunttasks/l10n-translate-verify.js]: done! @pdehaan

**Comment by vladikoff** [comment]
manually merged with output fixes https://github.com/mozilla/fxa-content-server/commit/dff66fa82cbc40813685bdc049a85837da7689d2

**Pull request closed without merging**